gpslogger/src/main/java/com/mendhak/gpslogger/common/Systems.java

```diff
@@ -173,5 +167,27 @@ static boolean hasUserGrantedPermission(String permissionName, Context context){
         return granted;
     }
 
+    public static boolean isDarkMode(FragmentActivity activity){
+        int nightModeFlags = activity.getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
+        return nightModeFlags == Configuration.UI_MODE_NIGHT_YES;
+    }
+
+    public static void setAppTheme(String appThemeSetting){
+
+        if(appThemeSetting.equalsIgnoreCase("system")){
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
+            } else {
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY);
```

Comment:
I've noticed sometimes going into battery saver mode is causing an application crash.  I cannot reliably reproduce it, but  it specifically happens on the API 28 emulator, with `MODE_NIGHT_AUTO_BATTERY`.  

It crashes with the error, "Context.startForegroundService() did not then call Service.startForeground()". 

Some explanation: a light/dark mode change causes Activities to restart, and that's not normally a problem.  But, because GPSLogger runs a ForegroundService, this was causing application crashes "Context.startForegroundService() did not then call Service.startForeground()". But, I _did_ call startForeground within 5 seconds, it's still crashing; there is some kind of edge case being run into where the OS is losing track of what was created.  [The issue is either unacknowledged or considered by design](https://issuetracker.google.com/issues/76112072). 

The only way around this problem could be to handle the configuration change and prevent the activity from restarting.  That means it will _require_ reopening the application to see the light/dark mode get applied.   It could be done by setting `android:configChanges="uiMode"` but I'll only do this if it really becomes a problem.  For now I'll let it stay dynamic.